packages/pipeline-editor/src/SubmissionHandler.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2018-2020 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  FrontendServices,
+  IDictionary,
+  RequestHandler
+} from '@elyra/application';
+import { showDialog, Dialog } from '@jupyterlab/apputils';
+import * as React from 'react';
+
+export class SubmissionHandler {
+  static async getRuntimeImages(): Promise<IDictionary<string>> {
+    const runtimeImages = await FrontendServices.getMetadata('runtime-images');
+    const images: { [key: string]: string } = {};
+    for (const image in runtimeImages) {
+      const imageName: string = runtimeImages[image]['metadata']['image_name'];
+      images[imageName] = runtimeImages[image]['display_name'];
```

Comment:
I think using the `name` field would probably make for a better index than `image_name`.  `name` is derived from the [instance's filename](https://github.com/elyra-ai/elyra/tree/master/etc/config/metadata/runtime-images), whereas `image_name` may contain ["unfriendly" characters](https://github.com/elyra-ai/elyra/blob/master/etc/config/metadata/runtime-images/tensorflow_2x_gpu_py3.json#L4).

Reply:
Sadly we didn't get to pick the index. The json returned here is actually copied as is into a canvas schema. So it has to match the expected format :(

Reply:
addendum to my previous comment, we do have a function that inserts the json into that schema file so we could always make this cleaner and then "revert" is to the necessary format in that function. If you really feel the service function should return a better formatted dict.

Reply:
Could you please provide a reference to the schema file?  I'm sure it's from https://github.com/elyra-ai/pipeline-schemas but never know which is what in there.

Changing the index sounds like more effort than its worth since we don't really control things here.  I figured `images[]` was our thing.

Reply:
This is the template we insert it into https://github.com/elyra-ai/elyra/blob/1947597c3b3e63e7377ac9bffae6f1c083f2fe9d/packages/pipeline-editor/src/pipeline-template.json

Reply:
So the _keys_ of `images` wind up filling in the [`{{image}}`](https://github.com/elyra-ai/elyra/blob/1947597c3b3e63e7377ac9bffae6f1c083f2fe9d/packages/pipeline-editor/src/pipeline-template.json#L16) in the template?

And does `display_name` get used to map to the image in a dropdown somewhere?

Reply:
whoops that was the wrong file https://github.com/elyra-ai/elyra/blob/87f864b6951c1fc5281ccfd9166e7f5945098825/packages/pipeline-editor/src/properties.json